model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -457,21 +480,86 @@ def _parse_outputs(self, metrics):
             mrc.perf_config() for mrc in self._config.model_run_configs()
         ]:
-            # Remove the latency file and all associated composing model latency files
+            # Remove the latency/profile export files and all associated composing model latency files
             for f in glob.glob(f"*{perf_config['latency-report-file']}"):
                 os.remove(f)
+            for f in glob.glob(f"*{perf_config['profile-export-file']}"):
+                os.remove(f)
+
+    def _extract_gpu_records(self, perf_config, metrics):
+        if perf_config["profile-export-file"]:
+            return
+
+        with open(perf_config["latency-report-file"], mode="r") as f:
+            csv_reader = csv.DictReader(f, delimiter=",")
+
+            for row in csv_reader:
+                self._perf_records[
+                    perf_config["model-name"]
+                ] = self._extract_perf_records_from_row(metrics, row)
+                self._gpu_records = self._extract_gpu_records_from_row(metrics, row)
+
+    def _extract_llm_records(self, perf_config, metrics):
+        if not perf_config["profile-export-file"]:
+            return
+
+        self._llm_records[perf_config["model-name"]] = []
+
+        with open(perf_config["profile-export-file"], mode="r") as f:
+            llm_output = json.load(f)
+
+            avg_first_token_to_token_latency = (
+                self._calculate_avg_first_token_to_token_latency(llm_output)
+            )
+            record = PerfAnalyzer.llm_metric_table[0][PerfAnalyzer.RECORD_CLASS](
+                value=avg_first_token_to_token_latency
+            )  # type: ignore
+
+            self._llm_records[perf_config["model-name"]].append(record)
+
+            avg_token_to_token_latency = self._calculate_avg_token_to_token_latency(
+                llm_output
+            )
+            record = PerfAnalyzer.llm_metric_table[1][PerfAnalyzer.RECORD_CLASS](
+                value=avg_token_to_token_latency
+            )  # type: ignore
+            self._llm_records[perf_config["model-name"]].append(record)
+
+    def _calculate_avg_first_token_to_token_latency(self, llm_output: Dict) -> float:
+        total_first_token_latency = 0
+        for request in llm_output["experiments"][0]["requests"]:
+            total_first_token_latency += (
+                request["response_timestamps"][0] - request["timestamp"]
+            )
+
+        avg_first_token_to_token_latency = total_first_token_latency / len(
+            llm_output["experiments"][0]["requests"]
+        )
+
+        return avg_first_token_to_token_latency
+
+    def _calculate_avg_token_to_token_latency(self, llm_output: Dict) -> float:
+        total_token_latency = 0.0
+        for request in llm_output["experiments"][0]["requests"]:
+            total_response_latency = 0
+            for response_timestamp in request["response_timestamps"]:
+                total_response_latency += response_timestamp - request["timestamp"]
```

Comment:
The T2T latency calculation should be the difference between the two consecutive responses, not response - request timestamp. We can use `itertools.pairwise`:

```suggestion
            for prev_response, response in pairwise(request["response_timestamps"]):
                total_response_latency += response - prev_response
```

Reply:
Understand and I've made the fix. However, itertools.pairwise isn't available, so I've had to code this the old-fashioned way.